sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -162,12 +164,19 @@ public static boolean isNullOrEmpty(CharSequence charSequence) {
 
 
     /**
-     * Helper method that returns an immutable {@link Map} of properties defined in {@code propertiesFileName}.
+     * Helper method that returns {@link UserAgentProperties} from properties defined in {@code propertiesFileName}.
      *
      * @param propertiesFileName The file name defining the properties.
-     * @return an immutable {@link Map}.
+     * @return {@link UserAgentProperties}.
      */
-    public static Map<String, String> getProperties(String propertiesFileName) {
+    public static UserAgentProperties getUserAgentPropertiesFromProperties(String propertiesFileName) {
```

Comment:
Is this something that we would want in our public API?

Reply:
The change seems not publish yet. Will double check

Reply:
Or we can leave both here.